charts/kcp-operator/values.yaml

```diff
@@ -0,0 +1,98 @@
+# Default values for kcp-operator.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+crds:
+  enabled: true
+
+# This will set the replicaset count more information can be found here: https://kubernetes.io/docs/concepts/workloads/controllers/replicaset/
+replicaCount: 1
+
+# This sets the container image more information can be found here: https://kubernetes.io/docs/concepts/containers/images/
+image:
+  repository: ghcr.io/kcp-dev/kcp-operator
+  # This sets the pull policy for images.
+  pullPolicy: IfNotPresent
+  # Overrides the image tag whose default is the chart appVersion.
+  tag: ""
+
+# This is for the secretes for pulling an image from a private repository more information can be found here: https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/
+imagePullSecrets: []
+# This is to override the chart name.
+nameOverride: ""
+fullnameOverride: ""
+
+# This section builds out the service account more information can be found here: https://kubernetes.io/docs/concepts/security/service-accounts/
+serviceAccount:
+  # Specifies whether a service account should be created
+  create: true
+  # Automatically mount a ServiceAccount's API credentials?
+  automount: true
+  # Annotations to add to the service account
+  annotations: {}
+  # The name of the service account to use.
+  # If not set and create is true, a name is generated using the fullname template
+  name: ""
+
+# This configures the operator's leader election to determine the Pod that is actively reconciling objects.
+leaderElection:
+  enabled: true
+
+# This is for setting Kubernetes Labels to the kcp-operator Deployment.
+labels: {}
+# This is for setting Kubernetes Annotations to the kcp-operator Deployment.
+annotations: {}
+
+# This is for setting Kubernetes Annotations to a Pod.
+# For more information checkout: https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/
+podAnnotations: {}
+# This is for setting Kubernetes Labels to a Pod.
+# For more information checkout: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/
+podLabels: {}
+
+podSecurityContext:
+  fsGroup: 65532
+
+securityContext:
+  capabilities:
+    drop:
+    - ALL
+  readOnlyRootFilesystem: true
+  runAsNonRoot: true
+  runAsUser: 65532
```

Comment:
Is this really configurable? Doesn't most of this depend on the operator and the container image we're building? What happens if someone just changes the runAsUser value? Won't this explode if it doesn't match the image?

Reply:
This should stay configurable because with the `kcp` chart we had the challenge that you couldn't run it on OpenShift when we changed security Context settings. So yes, if you fiddle with these settings the image might explode, but it's one of those "if you know what you are doing (or are on OpenShift), feel free to fiddle with it" advanced settings we need to have.